pkg/cmd/build/build.go

```diff
@@ -0,0 +1,421 @@
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(options.IidFile, []byte(id), 0600); err != nil {
+			return err
+		}
+	}
+
+	if len(tags) > 1 {
+		client, ctx, cancel, err := clientutil.NewClient(ctx, options.GOptions.Namespace, options.GOptions.Address)
+		logrus.Debug("Found more than 1 tag")
+		if err != nil {
+			return fmt.Errorf("unable to tag images: %s", err)
+		}
+		defer cancel()
+		imageService := client.ImageService()
+		image, err := imageService.Get(ctx, tags[0])
+		if err != nil {
+			return fmt.Errorf("unable to tag image: %s", err)
+		}
+		for _, targetRef := range tags[1:] {
+			image.Name = targetRef
+			if _, err := imageService.Create(ctx, image); err != nil {
+				// if already exists; skip.
+				if errors.Is(err, errdefs.ErrAlreadyExists) {
+					continue
+				}
+				return fmt.Errorf("unable to tag image: %s", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// TODO: This struct and `loadImage` are duplicated with the code in `cmd/load.go`, remove it after `load.go` has been refactor
+type readCounter struct {
+	io.Reader
+	N int
+}
+
+func loadImage(ctx context.Context, in io.Reader, namespace, address, snapshotter string, output io.Writer, platMC platforms.MatchComparer, quiet bool) error {
+	// In addition to passing WithImagePlatform() to client.Import(), we also need to pass WithDefaultPlatform() to NewClient().
+	// Otherwise unpacking may fail.
+	client, ctx, cancel, err := clientutil.NewClient(ctx, namespace, address, containerd.WithDefaultPlatform(platMC))
+	if err != nil {
+		return err
+	}
+	defer cancel()
+
+	r := &readCounter{Reader: in}
+	imgs, err := client.Import(ctx, r, containerd.WithDigestRef(archive.DigestTranslator(snapshotter)), containerd.WithSkipDigestRef(func(name string) bool { return name != "" }), containerd.WithImportPlatform(platMC))
+	if err != nil {
+		if r.N == 0 {
+			// Avoid confusing "unrecognized image format"
+			return errors.New("no image was built")
+		}
+		if errors.Is(err, images.ErrEmptyWalk) {
+			err = fmt.Errorf("%w (Hint: set `--platform=PLATFORM` or `--all-platforms`)", err)
+		}
+		return err
+	}
+	for _, img := range imgs {
+		image := containerd.NewImageWithPlatform(client, img, platMC)
+
+		// TODO: Show unpack status
+		if !quiet {
+			fmt.Fprintf(output, "unpacking %s (%s)...\n", img.Name, img.Target.Digest)
+		}
+		err = image.Unpack(ctx, snapshotter)
+		if err != nil {
+			return err
+		}
+		if quiet {
+			fmt.Fprintln(output, img.Target.Digest)
+		} else {
+			fmt.Fprintf(output, "Loaded image: %s\n", img.Name)
+		}
+	}
+
+	return nil
+}
+
+func generateBuildctlArgs(ctx context.Context, stdin io.Reader, options *types.BuildCommandOptions) (buildCtlBinary string,
+	buildctlArgs []string, needsLoading bool, metaFile string, tags []string, cleanup func(), err error) {
+
+	buildctlBinary, err := buildkitutil.BuildctlBinary()
+	if err != nil {
+		return "", nil, false, "", nil, nil, err
+	}
+
+	buildContext := options.BuildContext
```

Comment:
it's used later in line 247, suggest move there and use `options.buildContext` directly.